mythril/laser/ethereum/state/constraints.py

```diff
@@ -1,77 +1,98 @@
 """This module contains the class used to represent state-change constraints in
 the call graph."""
 
+from mythril.laser.smt import Solver, Bool
+
+from typing import List, Optional
+from z3 import unsat
+
 
 class Constraints(list):
     """This class should maintain a solver and it's constraints, This class
     tries to make the Constraints() object as a simple list of constraints with
     some background processing.
 
-    TODO: add the solver to this class after callback refactor
     """
 
-    def __init__(self, constraint_list=None, solver=None, possibility=None):
+    def __init__(
+        self,
+        constraint_list: Optional[List[Bool]] = None,
+        is_possible: Optional[bool] = None,
+    ) -> None:
         """
 
-        :param constraint_list:
-        :param solver:
-        :param possibility:
+        :param constraint_list: List of constraints
+        :param is_possible: Whether it is possible to satisfy the constraints or not
         """
         super(Constraints, self).__init__(constraint_list or [])
-        self.solver = solver
-        self.__possibility = possibility
+        self.__default_timeout = 10
+        self.__is_possible = is_possible
```

Comment:
Why are these variables defined with double underscore?

Reply:
They are super secret variables which can't even be acccesed by obj.__<name>

Reply:
^^, isn't that already the case if you do _
As I understand it, this would only make sure that classes inheriting this class would also not be able to access the variable